### PR TITLE
Find similar incidents bug

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33927

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by guykeller** [comment]
@ChanochShayner 
A few notes:
1. Did you notice the build is failing?
2. Is there a former build that failed on the unit tests, and passes now? it is the best way for us to know the bug was solved

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 1]
```diff
@@ -108,16 +109,30 @@ def get_incident_labels_map(labels):
     return labels_map
 
 
+def build_similar_keys_list(similar_incident_keys):
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:112]: How are the keys similar?
We should consider changing the name and the name of the parameter to something more clear

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 2]
```diff
@@ -108,16 +109,30 @@ def get_incident_labels_map(labels):
...
+    for key, value in similar_incident_keys.items():
+        if isinstance(value, int):
+            value = str(value).replace('"', r'\"').replace("\n", "\\n").replace("\r", "\\r")
+            compare_str = '{}:="{}"'
+            similar_key = compare_str.format(key, value)
+            similar_keys_list.append(similar_key)
+
+        else:
+            value = value.replace('"', r'\"').replace("\n", "\\n").replace("\r", "\\r")
+            value = value.encode('utf-8')
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:123]: 1. Can you do this line before the previous one?
If so, we have 4 lines here that are exactly the same, and can be done once.

2. Why don't we solely change the comparison string based on the type?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py (hunk 1 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py]: done

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py (hunk 2 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:122]: 1. no, Right now all the lines are different from each other.
2. done

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 3]
```diff
@@ -108,18 +109,26 @@ def get_incident_labels_map(labels):
     return labels_map
 
 
+def build_incident_fields_query(incident_data):
+    similar_keys_list = []
+    for key, value in incident_data.items():
+        str_value = str(value) if isinstance(value, int) else value
+        str_value = str_value.replace('"', r'\"').replace("\n", "\\n").replace("\r", "\\r")
+        str_value = str_value.encode('utf-8') if not isinstance(value, int) else str_value
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:117]: Let's extract to an external function with a readable name

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 4]
```diff
@@ -108,18 +109,26 @@ def get_incident_labels_map(labels):
...
+        similar_key = '{}:="{}"'.format(key, str_value) if isinstance(value, int) else '{}="{}"'.format(key, str_value)
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:118]: Let's split into two lines, with '{}:="{}"' and '{}="{}"' in a one-line if and a new variable

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py (hunk 3 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py]: done

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py (hunk 4 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py]: done

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/find_similar_incidents_test.py [hunk 5]
```diff
@@ -181,6 +181,16 @@ def test_get_incidents_by_keys():
     assert res == incident_by_keys
 
 
+def test_build_similar_keys_list():
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/find_similar_incidents_test.py:184]: Let's add a given-when-then here.
Other than that - approved

**Review by guykeller — APPROVED** [review]
see my comment

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/find_similar_incidents_test.py (hunk 5 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/find_similar_incidents_test.py:184]: dnoe

File: Packs/CommonScripts/ReleaseNotes/1_3_23.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+#### Scripts
+##### FindSimilarIncidents
+- Fixed an issue where the script did not find similar incidents when using **similarIncidentFields** argument from type int.
```
Comment (amshamah419) [inline Packs/CommonScripts/ReleaseNotes/1_3_23.md:4]:
```suggestion
- Fixed an issue where the script would not find similar incidents while using the **similarIncidentFields** argument when given an int type.
```

**Pull request merged**